shell/renderer/renderer_client_base.cc

```diff
@@ -483,11 +514,78 @@ bool RendererClientBase::IsWebViewFrame(
 
   gin_helper::Dictionary frame_element_dict(isolate, frame_element);
 
-  v8::Local<v8::Object> internal;
-  if (!frame_element_dict.GetHidden("internal", &internal))
-    return false;
+  bool is_webview = false;
+  return frame_element_dict.GetHidden("isWebView", &is_webview) && is_webview;
+}
+
+void RendererClientBase::SetupMainWorldOverrides(
+    v8::Handle<v8::Context> context,
+    content::RenderFrame* render_frame) {
+  auto prefs = render_frame->GetBlinkPreferences();
+  // We only need to run the isolated bundle if webview is enabled
+  if (!prefs.webview_tag)
+    return;
+
+  // Setup window overrides in the main world context
+  // Wrap the bundle into a function that receives the isolatedApi as
+  // an argument.
+  auto* isolate = context->GetIsolate();
+
+  gin_helper::Dictionary isolated_api = gin::Dictionary::CreateEmpty(isolate);
+  isolated_api.SetMethod("allowGuestViewElementDefinition",
+                         AllowGuestViewElementDefinition);
+  isolated_api.SetMethod("getWebFrameId", GetWebFrameId);
+  isolated_api.SetMethod("setIsWebView", SetIsWebView);
+  isolated_api.SetMethod("createNativeImage", &api::NativeImage::CreateEmpty);
+
+  std::vector<v8::Local<v8::String>> isolated_bundle_params = {
+      node::FIXED_ONE_BYTE_STRING(isolate, "isolatedApi")};
+
+  std::vector<v8::Local<v8::Value>> isolated_bundle_args = {
+      isolated_api.GetHandle()};
+
+  util::CompileAndCall(context, "electron/js2c/isolated_bundle",
+                       &isolated_bundle_params, &isolated_bundle_args, nullptr);
+}
+
+// static
+void RendererClientBase::AllowGuestViewElementDefinition(
+    v8::Isolate* isolate,
+    v8::Local<v8::Object> context,
+    v8::Local<v8::Function> register_cb) {
+  v8::HandleScope handle_scope(isolate);
+  v8::Context::Scope context_scope(context->CreationContext());
+  blink::WebCustomElement::EmbedderNamesAllowedScope embedder_names_scope;
+
+  content::RenderFrame* render_frame = GetRenderFrame(context);
+  RenderFrameStatus render_frame_status(render_frame);
+
+  if (!render_frame_status.is_ok())
+    return;
```

Comment:
This seems like an overly complicated way to write
```suggestion
  if (!render_frame)
    return;
```

Reply:
I've just moved the code from here without modification
https://github.com/electron/electron/blob/dca3b41ee6633718e0a541aa4c2f5a44c608b242/shell/renderer/api/electron_api_web_frame.cc#L562-L579

Reply:
It was added by @zcbenz in #13869

Reply:
The code was copied from https://source.chromium.org/chromium/chromium/src/+/main:extensions/renderer/guest_view/guest_view_internal_custom_bindings.cc;l=164-176?q=RenderFrameStatus&ss=chromium

According to the comment we can remove `RenderFrameStatus` in our code since we are not using `render_frame` after executing user code.